src/components/VAudioTrack/VWaveform.vue

```diff
@@ -332,8 +331,13 @@ export default defineComponent({
     const peakCount = computed(() =>
       getPeaksInWidth(waveformDimens.value.width)
     )
+    const peaks = computed(() =>
+      props.peaks?.length
+        ? props.peaks
+        : Array.from({ length: 100 }, () => Math.random())
+    )
```

Comment:
In a comment elsewhere you mentioned moving the logic up into `decode*`. I liked that suggestion because it alleviated the component from having to handle edge cases, essentially moving the edge-case handling "up the chain". Other places that rely on the waveform in the future would need to also handle this edge-case.

Reply:
I picked this because decode currently has no media-type specific logic.